lib/chef/knife/bootstrap/templates/chef-full.erb

```diff
@@ -1,14 +1,33 @@
-bash -c '
+sh -c '
 <%= "export https_proxy=\"#{knife_config[:bootstrap_proxy]}\"" if knife_config[:bootstrap_proxy] -%>
 
-distro=`uname -s`
+if test "x$TMPDIR" = "x"; then
+  tmp="/tmp"
+else
+  tmp=$TMPDIR
+fi
+
+# secure-ish temp dir creation without having mktemp available (DDoS-able but not exploitable)
+tmp_dir="$tmp/install.sh.$$"
+(umask 077 && mkdir $tmp_dir) || exit 1
 
-if test "x$distro" = "xSunOS"; then
-  if test -d "/usr/sfw/bin"; then
-    PATH=/usr/sfw/bin:$PATH
-    export PATH
+# @param $1 the omnibus root directory
+# @param $2 the requested version of omnibus package
```

Comment:
Yeah, getting idempotency here is likely to be significantly more complicated that it first appears.  And IMO if you're running a knife bootstrap, you expect an install to happen.